File: src/main/java/com/jiro4989/tkfm/model/ImageFilesModel.java
package com.jiro4989.tkfm.model;

import java.io.File;
import java.util.LinkedList;
import javafx.collections.FXCollections;
import javafx.collections.ObservableList;

public class ImageFilesModel {
  private ObservableList<ImageFileModel> files =
      FXCollections.observableArrayList(new LinkedList<>());
  private final CroppingImageModel croppingImage;

  public ImageFilesModel(CroppingImageModel ci) {
    croppingImage = ci;
  }

  public void add(ImageFileModel file) {
    files.add(file);
    select(0);
  }

  public void add(File file) {
    add(new ImageFileModel(file));
  }

  public void remove(int i) {
    if (i < 0 || files.size() <= i) return;
    files.remove(i);
    select(i);
  }

  public void clear() {
    files.clear();
    croppingImage.clearImage();
  }

  public void select(int i) {
    int min = 0;
    int max = files.size();
    if (i < min || max <= i) {
      return;
    }

    var file = files.get(i);
    var img = file.readImage();
    croppingImage.setImage(img);
  }

  public ObservableList<ImageFileModel> getFiles() {
    return files;
  }
}

File: src/main/kotlin/com/jiro4989/tkfm/model/ImageFilesModel.kt
package com.jiro4989.tkfm.model

import java.io.File
import javafx.collections.FXCollections
import javafx.collections.ObservableList

data class ImageFilesModel(
    private val croppingImage: CroppingImageModel,
    val files: ObservableList<ImageFileModel> = FXCollections.observableArrayList(mutableListOf())
) {

  fun add(file: ImageFileModel) {
    files.add(file)
    select(0)
  }

  fun add(file: File) {
    add(ImageFileModel(file))
  }

  fun remove(i: Int) {
    if (i < 0 || files.size <= i) return
    files.removeAt(i)
    select(i)
  }

  fun clear() {
    files.clear()
    croppingImage.clearImage()
  }

  /** i番目の画像を読み込んでトリミング済み画像としてセットする */
  fun select(i: Int) {
    val min = 0
    val max = files.size
    if (i < min || max <= i) {
      return
    }

    val file = files[i]
    val img = file.readImage()
    croppingImage.setImage(img)
  }
}
